Va416x0/Drv/AdcSampler/AdcSampler.cpp

```diff
@@ -71,20 +78,42 @@ AdcSampler ::AdcSampler(const char* const compName) : AdcSamplerComponentBase(co
     this->m_curRequest = 0;
     this->m_numReads = 0;
     this->m_requestIdx.store(0);
+    this->m_adcDelayTicks = 0;
 }
 
-void AdcSampler ::setup(AdcConfig& config, U32 interrupt_priority) {
+void AdcSampler ::setup(AdcConfig& config,
+                        U32 interrupt_priority,
+                        U32 adc_delay_microseconds,
+                        U8 timer_peripheral_index) {
+    Va416x0Mmio::Timer timer(timer_peripheral_index);
+    this->m_timer = timer;
```

Comment:
You might consider accepting `Va416x0Mmio::Timer` as an argument, for better type checking of the usages of this interface.